pkg/handlers/decide.go

```diff
@@ -18,23 +18,27 @@
 package handlers
 
 import (
+	"errors"
 	"net/http"
 
 	"github.com/optimizely/agent/pkg/middleware"
 
 	"github.com/optimizely/go-sdk/pkg/client"
 	"github.com/optimizely/go-sdk/pkg/decide"
 	"github.com/optimizely/go-sdk/pkg/decision"
+	"github.com/optimizely/go-sdk/pkg/odp/segment"
 
 	"github.com/go-chi/render"
 )
 
 // DecideBody defines the request body for decide API
 type DecideBody struct {
-	UserID          string                 `json:"userId"`
-	UserAttributes  map[string]interface{} `json:"userAttributes"`
-	DecideOptions   []string               `json:"decideOptions"`
-	ForcedDecisions []ForcedDecision       `json:"forcedDecisions,omitempty"`
+	UserID               string                            `json:"userId"`
+	UserAttributes       map[string]interface{}            `json:"userAttributes"`
+	DecideOptions        []string                          `json:"decideOptions"`
+	ForcedDecisions      []ForcedDecision                  `json:"forcedDecisions,omitempty"`
+	FetchSegments        bool                              `json:"fetchSegments"`
+	FetchSegmentsOptions []segment.OptimizelySegmentOption `json:"fetchSegmentsOptions,omitempty"`
```

Comment:
Can we confirm that if user sends a wrong value here then this will not break? maybe have 3 unit tests:
1. pass a single incorrect value in array
2. pass a mixture of correct and incorrect values
3. pass a empty array